packages/create-sitecore-jss/src/templates/nextjs/src/pages/[[...path]].tsx

```diff
@@ -29,9 +30,11 @@ const SitecorePage = ({ notFound, componentProps, layoutData }: SitecorePageProp
     return <NotFound />;
   }
 
+  const isEditing = layoutData.sitecore.context.pageEditing;
+
   return (
     <ComponentPropsContext value={componentProps}>
-      <SitecoreContext componentFactory={componentFactory} layoutData={layoutData}>
+      <SitecoreContext componentFactory={isEditing? editingComponentFactory:componentFactory} layoutData={layoutData}>
```

Comment:
Something wrong with the indent, please, run `lint` when you add changes, or run `prettier` which is configured